packages/frontend-shared/src/gql-components/ChooseExternalEditor.vue

```diff
@@ -0,0 +1,164 @@
+<template>
+  <div class="flex items-center">
+    <input
+      v-model="editorToUse"
+      type="radio"
+      class="mr-5px"
+      data-cy="use-well-known-editor"
+      value="found"
+      @change="saveEditor"
+    >
+
+    <Select
+      :model-value="selectedWellKnownEditor"
+      :options="externalEditors"
+      item-value="name"
+      item-key="id"
+      :placeholder="t('settingsPage.editor.noEditorSelectedPlaceholder')"
+      class="w-400px"
+      @update:model-value="updateEditor"
+    >
+      <template #input-prefix="{ value }">
+        <Icon
+          v-if="value"
+          :icon="icons[value.id]"
+          class="text-md"
+        />
+        <Icon
+          v-else
+          :icon="IconTerminal"
+          class="text-gray-600 text-md"
+        />
+      </template>
+      <template #item-prefix="{ value }">
+        <Icon
+          :icon="value.icon"
+          class="text-md"
+        />
+      </template>
+    </Select>
+  </div>
+
+  <div class="flex items-center py-2">
+    <input
+      v-model="editorToUse"
+      type="radio"
+      class="mr-5px"
+      value="custom"
+      data-cy="use-custom-editor"
+      @change="saveEditor"
+    >
+
+    <div class="w-400px">
+      <Input
+        v-model="customBinary"
+        input-classes="text-sm"
+        placeholder="Custom path..."
+      >
+        <template #prefix>
+          <Icon
+            :icon="IconTerminal"
+            class="text-gray-600 text-md"
+          />
+        </template>
+      </Input>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ref, computed, watch, FunctionalComponent, SVGAttributes } from 'vue'
+import Icon from '@cy/components/Icon.vue'
+import { useI18n } from '@cy/i18n'
+import Select from '@cy/components/Select.vue'
+import Input from '@cy/components/Input.vue'
+import VSCode from '~icons/logos/visual-studio-code'
+import Atom from '~icons/logos/atom-icon'
+import Webstorm from '~icons/logos/webstorm'
+import Vim from '~icons/logos/vim'
+import Sublime from '~icons/logos/sublimetext-icon'
+import Emacs from '~icons/logos/emacs'
+import IconTerminal from '~icons/mdi/terminal'
```

Comment:
Why are those icons not part of the `env-editors.ts` ?

Reply:
I get it, because they are not available there.

Maybe we could use import meta to import a glob of logos and only reference logos by name on the server...

This is an architecture concern though.

Reply:
Could we at least add one icon corresponding to `computer` for the open in file system choice.

And use the terminal icon from cypress instead of MDI?

Reply:
Done and done for icons, let's consider the Vite globbing, too, at some point...

![image](https://user-images.githubusercontent.com/19196536/142793578-7c71bfd2-72b8-405f-8fc7-02f9db78b5f3.png)

Reply:
love it